code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -114,8 +111,9 @@ GLOBAL_DATUM_INIT(latejoin_menu, /datum/latejoin_menu, new)
 
 	return list("departments_static" = departments)
 
-/datum/latejoin_menu/ui_state(mob/user)
-	return GLOB.new_player_state
+// we can't use GLOB.new_player_state here since it also allows any admin to see the ui, which will cause runtimes
```

Comment:
its 1 am i meant to say that the UI won't close if an admin tries to latejoin with `new_player_state`